woocommerce/woocommerce-gutenberg-products-block
# Convert shipping selection Checkout tests to playwright

This PR moves the following tests to playwright and removes the puppeteer equivalent:

- User can choose free shipping
- User can choose flat rate shipping

It also creates a "Checkout page object" util class which contains some new utils, specifically:

- `selectAndVerifyShippingOption`
  - `isShippingRateSelected`
- `checkCustomerPushCompleted`
- `fillInCheckoutWithTestData`
  - `fillShippingDetails`
  - `fillBillingDetails`
- `placeOrder`
- `verifyAddressDetails`

These are re-makes of the existing puppeteer ones from `tests/utils/shopper.js` with some improvements:

- Use text to get elements instead of CSS selectors/IDs where possible
- Remove specific `waitFor...` functions from tests, we get this for free with playwright
- Remove the interval from `checkIfShippingHasChanged` and instead assert whether the API response is complete, then check the shipping section of the sidebar to assert if it updated.
- Improve `fillInCheckoutWithTestData` to check if billing and shipping is open, and fill the ones that are open.

This `pageObject` is injected into each test by extending the `test` object from `@woocommerce/e2e-playwright-utils`.


It also adds an `emptyCart` function to `frontendUtils` which removes all items from the cart.

#### Other Checks


### Testing

#### Automated Tests

#### User Facing Testing

No testing required, just ensure Playwright E2E tests pass and ensure Puppeteer E2Es also pass, so that we can be sure the removed tests don't impact on anything.


### WooCommerce Visibility




### Performance Impact



### Changelog

> Skipping
